src/OpenTelemetry.Api/Context/Propagation/TraceContextFormat.cs

```diff
@@ -81,6 +82,23 @@ public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> ge
         /// <inheritdoc/>
         public void Inject<T>(SpanContext spanContext, T carrier, Action<T, string, string> setter)
         {
+            if (spanContext == null || !spanContext.IsValid)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectSpanContext("Invalid context");
+            }
```

Comment:
Should we return early when the spanContext is invalid like we do if carrier and setter are null?